eng/Versions.props

```diff
@@ -44,6 +44,11 @@
     <!-- Targeting packs are only patched in extreme cases. -->
     <ProjectServicingConfiguration Include="Microsoft.NETCore.App.Ref" PatchVersion="0" />
   </ItemGroup>
+  <PropertyGroup>
+    <!-- For source generator support -->
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>3.10.0-1.final</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
```

Comment:
I noticed this moved from `3.8` previously. Are we OK with depending on this new of a compiler? I see it was only released a month ago.

Reply:
I think @layomia would be needing it for his work anyway

Reply:
I only upgraded so I could use the new `GeneratorPostInitializationContext` feature to support pre-generating the `JsonSerializableAttribute` (https://github.com/dotnet/runtimelab/pull/877) into user assemblies. For now, we are shipping the attribute from STJ.dll itself so we can stay on `3.8`. `3.9` is the minimum version needed to use the new feature.